molgenis-data/src/main/java/org/molgenis/data/decorator/NotifyingDecorator.java

```diff
@@ -0,0 +1,109 @@
+package org.molgenis.data.decorator;
+
+import org.molgenis.data.AbstractRepositoryDecorator;
+import org.molgenis.data.Entity;
+import org.molgenis.data.Repository;
+import org.molgenis.data.support.QueryImpl;
+
+import java.util.function.Consumer;
+import java.util.stream.Stream;
+
+import static java.util.Objects.requireNonNull;
+
+/**
+ * Decorator that notifies a listener of mutations that happen to the decorated repository.
+ *
+ * @param <E> the class of the entities in the repository
+ */
+public class NotifyingDecorator<E extends Entity> extends AbstractRepositoryDecorator<E>
+{
+	private final RepositoryListener<E> repositoryListener;
+
+	public NotifyingDecorator(Repository<E> delegateRepository, RepositoryListener<E> repositoryListener)
```

Comment:
I thought I was going to need this one but I didn't.
It'll be very useful for future decorators and probably also to simplify the existing row level security decorators so I've left it in.

Reply:
Lets leave unused code out and put it on a branch/gist for reintroduction once we start using it.